### PR TITLE
TC_00.002.15-A | New Item > Create Pipeline Project | The URL of the configure page contains new project name

### PR DESCRIPTION
**Pull request by LanaUno**
[TC_00.002.15](https://github.com/RedRoverSchool/JenkinsQA_JS_2024_fall/issues/516) | New Item > Create Pipeline Project | The URL of the configure page contains new project name

**Implemented Changes:**

This is newly created TC ( not refactoring) in POM format.

to **NewJobPage:**

- added locator getUrlConfigurePageField
- added locator getBreadcrumbsListItem

### CODE REVIEW

**Review by Negrescu29 — COMMENTED** [review]
please follow the instructions on what info a pr should have: https://docs.google.com/presentation/d/1cE3K9dxm0qO1eKg9JrGNpME8cdHLCgZ1/edit#slide=id.p50

**Comment by LanaUno** [comment]
> please follow the instructions on what info a pr should have: https://docs.google.com/presentation/d/1cE3K9dxm0qO1eKg9JrGNpME8cdHLCgZ1/edit#slide=id.p50

_Thank you for reminding, all necessary info added_

File: cypress/e2e/newItemCreatePipelineProjectPOM.cy.js [hunk 1]
```diff
@@ -88,4 +88,15 @@ describe("US_00.002 | New Item > Create Pipeline Project", () => {
...
+    dashboardPage.clickNewItemMenuLink();
+    newJobPage
+      .typeNewItemName(project.name)
+      .selectPipelineProject()
+      .clickOKButton()
+
+      .getBreadcrumbsListItem().should('have.text', 'Configuration')
+    newJobPage  
+      .getUrlConfigurePageField().should('include', project.name)
+  })
```
Comment (ivanlashkevich) [inline cypress/e2e/newItemCreatePipelineProjectPOM.cy.js:101]: A couple of things for potential test case improvement require your attention:
- your PR title should be the same as the TC name. So, it should be like in the RC card: 'TC_00.002.15-A | New Item > Create Pipeline Project | The URL of the configure page contains new project name';
- add PR link to your TC card;
-  for better readability and maintainability, the highlighted assertions in the test may be grouped like (there can be variations, but following your style):
```
newJobPage
      .getBreadcrumbsListItem().should('have.text', 'Configuration');
newJobPage
      .getUrlConfigurePageField().should('include', project.name);
```

**Review by ivanlashkevich — APPROVED** [review]

File: cypress/e2e/newItemCreatePipelineProjectPOM.cy.js (hunk 1 above)
Comment (LanaUno) [reply cypress/e2e/newItemCreatePipelineProjectPOM.cy.js:101]: > A couple of things for potential test case improvement require your attention:
> 
> * your PR title should be the same as the TC name. So, it should be like in the TC card: 'TC_00.002.15-A | New Item > Create Pipeline Project | The URL of the configure page contains new project name';
> * add PR link to your TC card;
> * for better readability and maintainability, the highlighted assertions in the test may be grouped like (there can be variations, but following your style):
> 
> ```
> newJobPage
>       .getBreadcrumbsListItem().should('have.text', 'Configuration');
> newJobPage
>       .getUrlConfigurePageField().should('include', project.name);
> ```

Thank you for collaborating, Ivan, PR title corrected and link is added, regarding code style I would leave like it is, I heard from Nalya several times that it is not necessary to repeat Class name all the time

File: cypress/pageObjects/NewJobPage.js [hunk 2]
```diff
@@ -68,6 +70,14 @@ class NewJobPage {
...
         return this;   
     }
+    
+     chooseRandomItemFromList() {
+        this.getAllItemsList().then((items) => {
+            const randomIndex = Math.floor(Math.random() * items.length);
+            cy.wrap(items).eq(randomIndex).click(); 
+        }).then(() => this);
+        return this;   
+    }
```
Comment (Negrescu29) [inline cypress/pageObjects/NewJobPage.js:80]: @LanaUno from what I can see this code is not used anywhere, is there a reason you added it?

File: cypress/e2e/newItemCreatePipelineProjectPOM.cy.js (hunk 1 above)
Comment (ivanlashkevich) [reply cypress/e2e/newItemCreatePipelineProjectPOM.cy.js:101]: I just paid your attention to this point, cause of separated assertions: you put one getter + assertion without specifying the page it relates and joined it to another getter + assertion, but this time you specify the page (e.g. newJobPage). This point underlines only the readability and maintainability of the test.

File: cypress/pageObjects/NewJobPage.js (hunk 2 above)
Comment (LanaUno) [reply cypress/pageObjects/NewJobPage.js:80]: This code is not mine, there was a conflict in my PR, I wrote you about it, those changes conflicted with mine, so I accepted them.

Comment (Negrescu29) [reply cypress/pageObjects/NewJobPage.js:80]: I'm not sure how you merged the conflict but this shouldn't show up as you're committing this code.

**Review by QA-LilyArslanova — APPROVED** [review]
Test passed

File: cypress/pageObjects/NewJobPage.js (hunk 2 above)
Comment (LanaUno) [reply cypress/pageObjects/NewJobPage.js:80]: > I'm not sure how you merged the conflict but this shouldn't show up as you're committing this code.

I see this code repeated twice, I will delete unnecessary lines in my next commit.

**Pull request merged**